### PR TITLE
ctl: add get_region_read_progress (release-6.5)

### PR DESCRIPTION
**Pull request by ekexium**
### What is changed and how it works?

Issue Number: ref #15082 

What's Changed:

Cherry pick #15099 

```commit-message
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

```release-note
Please add a release note.

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

If you don't think this PR needs a release note then fill it with None.
If this PR will be picked to release branch, then a release note is probably required.
```

### CODE REVIEW

File: components/server/src/server.rs [hunk 1]
```diff
@@ -268,9 +269,12 @@ struct TikvServer<ER: RaftEngine, F: KvFormat> {
     check_leader_worker: Worker,
     sst_worker: Option<Box<LazyWorker<String>>>,
     quota_limiter: Arc<QuotaLimiter>,
-    causal_ts_provider: Option<Arc<CausalTsProviderImpl>>, // used for rawkv apiv2
+    causal_ts_provider: Option<Arc<CausalTsProviderImpl>>,
+    // used for rawkv apiv2
```
Comment (MyonKeminta) [inline components/server/src/server.rs:273]: Emmm, should this commend be for the previous line?

File: components/server/src/server.rs [hunk 2]
```diff
@@ -268,9 +269,12 @@ struct TikvServer<ER: RaftEngine, F: KvFormat> {
...
     tablet_factory: Option<Arc<dyn TabletFactory<RocksEngine> + Send + Sync>>,
-    br_snap_recovery_mode: bool, // use for br snapshot recovery
+    br_snap_recovery_mode: bool,
+    // use for br snapshot recovery
```
Comment (MyonKeminta) [inline components/server/src/server.rs:276]: So do this

**Review by MyonKeminta — COMMENTED** [review]
Rest LGTM

File: components/server/src/server.rs (hunk 1 above)
Comment (ekexium) [reply components/server/src/server.rs]: This format tool is so weird

**Review by cfzjywxk — APPROVED** [review]

**Review by MyonKeminta — APPROVED** [review]

**Comment by ekexium** [comment]
@tikv/qa-release-merge PTAL

**Pull request merged**